include/eve/algo/views/iota.hpp

```diff
@@ -71,7 +70,7 @@ namespace eve::algo::views
     base_plus_offset_iterator& operator+=(std::ptrdiff_t n)
     {
       i += n;
-      wide_cur += scale * n;
+      wide_cur =  fam(wide_cur, scale, n);
```

Comment:
The reason this was not done with fam is that `scale * n` is optimised to a constant (if you iterate with the same step which most of the time you are

Reply:
Please compare asm

Reply:
You can see that this is wrong, right? We should multiply once - outside of the loop. And them just do add.
using fma prevents the correct codegen

Reply:
you are right, fma brings nothing as s*n is constant